Set RPATH for installed libraries for all non-system paths

simonspa opened the pull request:
With the current CMake setup, it is not possible to install delphes in a custom path since the libraries are not found by the executables due to the missing RPATH setting.

To demonstrate:
```
$ git clone git@github.com:delphes/delphes.git
$ cd delphes && mkddir build && cd build/
$ cmake -DCMAKE_INSTALL_PREFIX=../install ..
[...]
-- Build files have been written to: /home/simonspa/software/delphes/build
$ make -j8
[...]
[ 98%] Linking CXX executable DelphesSTDHEP
[ 99%] Linking CXX executable DelphesLHEF
[100%] Linking CXX executable DelphesHepMC
[100%] Linking CXX executable DelphesROOT
[100%] Built target DelphesSTDHEP
[100%] Built target DelphesLHEF
[100%] Built target DelphesHepMC
[100%] Built target DelphesROOT
[100%] Linking CXX executable Validation
[100%] Built target Validation
$ make install
Install the project...
[...]
-- Installing: /home/simonspa/software/delphes/install/lib/libDelphes.so
-- Set runtime path of "/home/simonspa/software/delphes/install/lib/libDelphes.so" to ""
-- Installing: /home/simonspa/software/delphes/install/lib/libDelphesDisplay.so
-- Set runtime path of "/home/simonspa/software/delphes/install/lib/libDelphesDisplay.so" to ""
$ cd ../install/bin/
$ ./DelphesSTDHEP ../cards/delphes_card_ILD.tcl output.root input.hep
./DelphesSTDHEP: error while loading shared libraries: libDelphes.so: cannot open shared object file: No such file or directory
```

After adding the RPATH configuration to CMake, the library paths are resolved without problems:
```
$ make install
-- Installing: /home/simonspa/software/delphes/install/lib/libDelphes.so
-- Set runtime path of "/home/simonspa/software/delphes/install/lib/libDelphes.so" to "/home/simonspa/software/delphes/install/lib:/opt/root5/lib"
-- Installing: /home/simonspa/software/delphes/install/lib/libDelphesDisplay.so
-- Set runtime path of "/home/simonspa/software/delphes/install/lib/libDelphesDisplay.so" to "/home/simonspa/software/delphes/install/lib:/opt/root5/lib"
$ cd ../install/bin
$ ./DelphesSTDHEP ../cards/delphes_card_ILD.tcl output.root input.hep
[...]
** Reading input.hep
** ERROR: can't open input.hep
```

Cheers,
Simon

pavel-demin commented:
Thanks for this pull request. I just have a couple of questions:
 - What is the reason to check isSystemDir as a string value? Would it be OK to replace `"${isSystemDir}" STREQUAL "-1"` with `${isSystemDir} EQUAL -1`?
 - Would it be OK if I make these settings optional and enabled by default? It would be probably useful to have a possibility to build relocatable binaries.

simonspa commented:
* I remember that I once had problems with the `EQUAL` at this place, but since it seems to be working fine with recent CMake versions I switched to comparing integers.
* Sure, there might be some useful cases for that. I added an option which defaults to `ON`

pavel-demin commented:
Great! Thanks for the updates and the answers.
